packages/bundle-source/src/fs.js

```diff
@@ -121,19 +121,23 @@ export const makeFileWriter = (
  *   path: Pick<import('path'), 'resolve' | 'relative' | 'normalize'>,
  * }} io
  * @param {number} pid
+ * @param {number} nonce
  * @param {(there: string) => ReturnType<makeAtomicFileWriter>} make
  */
 export const makeAtomicFileWriter = (
   fileName,
   { fs, path },
-  pid,
-  make = there => makeAtomicFileWriter(there, { fs, path }, pid, make),
+  pid = undefined,
+  nonce = undefined,
+  make = there => makeAtomicFileWriter(there, { fs, path }, pid, nonce, make),
 ) => {
+  const pidText = `${pid || 'no-pid'}`;
+  const nonceText = `${nonce || 'no-nonce'}`;
   const writer = makeFileWriter(fileName, { fs, path }, make);
   return harden({
     ...writer,
     atomicWriteText: async (txt, opts) => {
-      const scratchName = `${fileName}.${pid}.scratch`;
+      const scratchName = `${fileName}.${pidText}.${nonceText}.scratch`;
```

Comment:
Consider the following, to make globs easier `rm -rf *.19837.scratch`:

```suggestion
      const scratchName = `${fileName}.${nonceText}.${pidText}.scratch`;
```